### PR TITLE
Add support for reducible java.jdbc calls

### PR DESCRIPTION
**Pull request by senior**
Opening this PR as what I think is probably the most straightforward way to incorporate the reducible support that java.jdbc has added. Not sure if this is the best way, but it seemed like the easiest. Submitting this as an RFC of sorts. If this approach seems like the right way to go about it, I'll get the tests and docs written up to accompany it.

--------------------------------

Thanks for contributing to Toucan. Before open a pull request, please take a moment to:

- [x] Ensure the PR follows the [Clojure Style Guide](https://github.com/bbatsov/clojure-style-guide) and the [Metabase Clojure Style Guide](https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide).
- [x] Tests and linters pass. You can run them locally as follows:

      lein test && lein lint

    CircleCI will also run these same tests against your PR. See the [README](https://github.com/metabase/toucan#tests--linting) for more details on running tests.
- [x] Make sure you've included new tests for any new features or bugfixes
- [x] New features are documented, or documentation is updated appropriately for any changed features.
- [x] Carefully review your own changes and revert any superfluous ones. (A good example would be moving words in the Markdown documentation to different lines in a way that wouldn't change how the rendered page itself would appear. These sorts of changes make a PR bigger than it needs to be, and, thus, harder to review.)

    Of course, indentation and typo fixes are not covered by this rule and are always appreciated.
- [x] Include a detailed explanation of what changes you're making and why you've made them. This will help us understand what's going on while we review it.

Once you've done all that, open a PR! Make sure to at-mention @camsaul in the PR description. Otherwise I won't get an email about it and might not get review it right away. :)

Thanks for your contribution!

### CODE REVIEW

**Comment by camsaul** [comment]
Tests? :wink:

**Comment by camsaul** [comment]
In principle this looks good but we should add some tests & dox. Then we can merge

**Comment by senior** [comment]
@camsaul done

File: docs/db-functions.md [hunk 1]
```diff
@@ -229,6 +229,19 @@ syntax similar to the `select` family of functions, but returns a Boolean value:
...
+### select-reducible
+
+Using `select` will realize the full set of results in memory. For smaller sets of results this is fine but queries that
+could potentially return many rows, this could cause memory issues. `select-reducible` is an similar function to `select`,
+but will return a reducible sequence instead of a vector. Using this, it's possible to consume the query results as they
+are streamed from the database. Using this, you can avoid fully realizing the set of results in memory.
+
+```
+(transduce (filter complex-filter-logic-fn) serialize-to-http-response output-stream (db/select-reducible User :active true))
+```
```
Comment (camsaul) [inline docs/db-functions.md]: Can we 

1) Add `clj` after the triple-backtick to get syntax highlighting for the example
2) break the example up into multiple lines so we can see it without needing vertical scrollbars

Comment (camsaul) [reply docs/db-functions.md]: since everything else looks good I'll just go ahead and fix this

**Pull request merged**